examples/window-app/common/src/ZclCallbacks.cpp

```diff
@@ -39,83 +40,22 @@ using namespace ::chip::app::Clusters::WindowCovering;
-    }
-    else if (attributePath.mClusterId == Id)
-    {
-        ChipLogProgress(Zcl, "Window  cluster ID: " ChipLogFormatMEI, ChipLogValueMEI(attributePath.mClusterId));
-    }
-    else
-    {
-        ChipLogProgress(Zcl, "Unknown cluster ID: " ChipLogFormatMEI, ChipLogValueMEI(attributePath.mClusterId));
-    }
-
-    WindowApp & app     = WindowApp::Instance();
-    EndpointId endpoint = attributePath.mEndpointId;
-
-    chip::app::DataModel::Nullable<chip::Percent100ths> current;
-    chip::app::DataModel::Nullable<chip::Percent100ths> target;
-    OperationalState opState;
-
-    switch (attributePath.mAttributeId)
-    {
-    case Attributes::Type::Id:
-        app.PostEvent(WindowApp::Event(WindowApp::EventId::CoverTypeChange, endpoint));
         break;
-
-    case Attributes::CurrentPositionLiftPercent100ths::Id:
-        app.PostEvent(WindowApp::Event(WindowApp::EventId::LiftChanged, endpoint));
-        break;
-
-    case Attributes::CurrentPositionTiltPercent100ths::Id:
-        app.PostEvent(WindowApp::Event(WindowApp::EventId::TiltChanged, endpoint));
-        break;
-
-    case Attributes::TargetPositionLiftPercent100ths::Id:
-        Attributes::TargetPositionLiftPercent100ths::Get(endpoint, target);
-        Attributes::CurrentPositionLiftPercent100ths::Get(endpoint, current);
-        opState = ComputeOperationalState(target, current);
-        if (OperationalState::MovingDownOrClose == opState)
-        {
-            app.PostEvent(WindowApp::Event(WindowApp::EventId::LiftDown, endpoint));
-        }
-        else if (OperationalState::MovingUpOrOpen == opState)
-        {
-            app.PostEvent(WindowApp::Event(WindowApp::EventId::LiftUp, endpoint));
-        }
-        break;
-
-    case Attributes::TargetPositionTiltPercent100ths::Id:
-        Attributes::TargetPositionTiltPercent100ths::Get(endpoint, target);
-        Attributes::CurrentPositionTiltPercent100ths::Get(endpoint, current);
-        opState = ComputeOperationalState(target, current);
-        if (OperationalState::MovingDownOrClose == opState)
-        {
-            app.PostEvent(WindowApp::Event(WindowApp::EventId::TiltDown, endpoint));
-        }
-        else if (OperationalState::MovingUpOrOpen == opState)
-        {
-            app.PostEvent(WindowApp::Event(WindowApp::EventId::TiltUp, endpoint));
-        }
-        break;
-
     default:
         break;
     }
 }
 
-/**
- * @brief  Cluster StopMotion Command callback (from client)
- */
-bool emberAfWindowCoveringClusterStopMotionCallback(chip::app::CommandHandler * commandObj,
-                                                    const chip::app::ConcreteCommandPath & commandPath,
-                                                    const Commands::StopMotion::DecodableType & commandData)
+/* Forwards all attributes changes */
+void MatterWindowCoveringClusterServerAttributeChangedCallback(const app::ConcreteAttributePath & attributePath)
```

Comment:
This should be implemented in the cluster itself, no?

But if all it's doing is running app logic (for now), this should presumably just happen in the app post-attribute-changed callback.

Reply:
The post-attribute callback is forwarding everything in the queue for the EFR32 app demo . The other does simple fake motion with a timer to implement the Test . Further goals is to create a linux app and passing the test for TH in it rather than all-cluster-app